projects/ngx-skltn/src/lib/skltn/skltn.component.ts

```diff
@@ -106,21 +106,24 @@ export class SkltnComponent implements OnInit, OnDestroy, AfterViewInit {
     </style>`);
 
     // Update
-    this.updStream$.pipe(
-      debounceTime(100),
-    ).subscribe(() => this.calcShapes());
+    this.subscriptions = this.updStream$
+      .pipe(
+        debounceTime(100),
+      )
+      .subscribe(() => this.calcShapes());
 
     // Update href (Safari Bug, SVG Ref Path)
     this.href = window.location.href;
 
-    this.zone.runOutsideAngular(() => {
-      this.checkHrefIntervalID = window.setInterval(() => {
-        if (this.href !== window.location.href) {
-          this.href = window.location.href;
-          this.cd.detectChanges();
-        }
-      }, 100);
+    const checkHref = interval(100)
+      .subscribe(() => {
+          if (this.href !== window.location.href) {
+            this.href = window.location.href;
+            this.cd.detectChanges();
+          }
```

Comment:
Interval(100) will run Change Detection every 100ms, so you don't need `this.cd.detectChanges();`.
But we need this only when href changed to support Safari relative references.
So for performance reason we run this polling outside Zone.

Reply:
Creating the observable inside a runOutsideAngular call should do the trick then. I'm just not sure if there's any problem when unsubscribing inside the zone.